servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpExecutionStrategy.java

```diff
@@ -310,16 +284,10 @@ public boolean equals(final Object o) {
 
         final DefaultHttpExecutionStrategy that = (DefaultHttpExecutionStrategy) o;
 
-        if (offloads != that.offloads) {
-            return false;
-        }
-        if (threadAffinity != that.threadAffinity) {
-            return false;
-        }
-        if (!Objects.equals(executor, that.executor)) {
-            return false;
-        }
-        return mergeStrategy == that.mergeStrategy;
+        return offloads == that.offloads &&
```

Comment:
My preference is for exact types where there may be significant behavior differences between implementations and/or a different equals definition in other implementations. This would seem to be likely or possible with `HttpExecutionStrategy` implementations so I'd prefer to keep the current narrow definition. With #1695 it won't be possible as we won't control `equals()` and enum equals is always identity equals.

Reply:
Ok, I don't worry much about custom impls, just want to make sure that users won't see the difference in behavior if they pass `noOffloadsStrategy()` instead of `customStrategyBuilder().offloadNone().build()`. If that will be improved in #1695, I'm fine with that.